agent/ebs/watcher_test.go

```diff
@@ -300,6 +304,104 @@ func TestHandleDuplicateEBSAttachment(t *testing.T) {
+				mcc.EXPECT().NodeStageVolume(gomock.Any(),
+					taskresourcevolume.TestVolumeId, gomock.Any(),
+					filepath.Join(hostMountDir, taskresourcevolume.TestSourceVolumeHostPath),
+					taskresourcevolume.TestFileSystem, gomock.Any(), gomock.Any(),
+					gomock.Any(), gomock.Any(), gomock.Any()).Return(errors.New("test")).Times(1)
+			},
+			expectedErrorStubs:       []string{"Failed to initialize EBS volume ID"},
+			expectedAttachmentStatus: attachment.AttachmentNone,
+		},
+		{
+			name: "StageAll returns expected error when already attached",
+			setTaskStateExpectations: func(ts *mock_dockerstate.MockTaskEngineState) {
+				ebsResourceAttachment := getEBSResourceAttachment()
+				ebsResourceAttachment.SetAttachedStatus()
+				ts.EXPECT().GetEBSByVolumeId(taskresourcevolume.TestVolumeId).Return(ebsResourceAttachment, true).Times(2)
+			},
+			// we won't call NodeStage here
+			setCSIClientExpectations: func(mcc *mock_csiclient.MockCSIClient) {},
+			expectedErrorStubs:       nil,
+			expectedAttachmentStatus: attachment.AttachmentAttached,
+		},
+		{
+			name: "StageAll fails with missing volume ID",
+			setTaskStateExpectations: func(ts *mock_dockerstate.MockTaskEngineState) {
+				ts.EXPECT().GetEBSByVolumeId(taskresourcevolume.TestVolumeId).Return(nil, false).Times(1)
+				ebsResourceAttachment := getEBSResourceAttachment()
+				ts.EXPECT().GetEBSByVolumeId(taskresourcevolume.TestVolumeId).Return(ebsResourceAttachment, true).Times(1)
+			},
+			setCSIClientExpectations: func(mcc *mock_csiclient.MockCSIClient) {},
+			expectedErrorStubs:       []string{"Unable to find EBS volume with volume ID"},
+			expectedAttachmentStatus: attachment.AttachmentNone,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			ctx := context.Background()
+
+			taskEngineState := mock_dockerstate.NewMockTaskEngineState(mockCtrl)
+			tc.setTaskStateExpectations(taskEngineState)
+			mockDiscoveryClient := mock_ebs_discovery.NewMockEBSDiscovery(mockCtrl)
+			mockTaskEngine := mock_engine.NewMockTaskEngine(mockCtrl)
+			mockTaskEngine.EXPECT().GetDaemonTask(md.EbsCsiDriver).Return(nil).AnyTimes()
+			mockTaskEngine.EXPECT().GetDaemonManagers().Return(nil).AnyTimes()
+
+			mockCsiClient := mock_csiclient.NewMockCSIClient(mockCtrl)
+			tc.setCSIClientExpectations(mockCsiClient)
+
+			watcher := newTestEBSWatcher(ctx, taskEngineState, mockDiscoveryClient, mockTaskEngine, mockCsiClient)
+			foundVolumes := map[string]string{taskresourcevolume.TestVolumeId: taskresourcevolume.TestDeviceName}
+			errors := watcher.StageAll(foundVolumes)
+			assert.Equal(t, len(tc.expectedErrorStubs), len(errors))
+			if len(tc.expectedErrorStubs) != len(errors) {
+				t.Errorf("mismatched error count")
+			} else {
+				for i, err := range tc.expectedErrorStubs {
+					assert.True(t, strings.Contains(fmt.Sprintf("%s", errors[i]), err))
```

Comment:
nit: you can also use `assert.ErrorContains` instead of combining `assert.True` and `strings.Contains`.